### ISSUE TITLE
[IMPROVEMENT] Use upgrade-responder to store current list of kernel versions to avoid.

### ISSUE THREAD

**Issue by james-munson** [longhorn/longhorn#8042]
## Is your improvement request related to a feature? Please describe (👍 if you like this request)

After #7931, the script https://github.com/longhorn/longhorn/scripts/environment_check.sh checks kernel version against a list of known problematic releases, and so does node_controller.go, which will create a failed Condition in the nodes.longhorn.io CR if the cluster finds itself running on one.

The list of bad kernels is hard-coded, and exists in two different places.  (Three, counting the KB article https://github.com/longhorn/website/pull/873/files#diff-d41c88a85640725c3b7cbf4de27f9440d74122aebf2d3f34133f4ca71c18c5ea).

## Describe the solution you'd like

There should be code commonality that allows at least the environment check and the node controller condition check to use the same list.  That might happen in a flavor of the longhorn/pre-flight daemonset, whether as shared code in a common library or by using the daemonset as a permanent feature of a Longhorn installation.

Moreover, the list should be dynamically updated in a way that can be freshened without a code upgrade.  Ideally, it could be kept up-to-date in the upgrade-responder and fetched from there whenever the node condition check is run.

**Comment by derekbit** [comment]
Sounds good. Does it work in an air-gapped environment?

**Comment by james-munson** [comment]
> Sounds good. Does it work in an air-gapped environment?

I don't see how it could pull from the upgrade-responder (also a problem in general for air-gapped installations.)  But it might be possible to include a file with a copy of the most recent list in the list of artifacts to be supplied when an air-gapped installation is to be done.  That would be a bit better than having to use the most recent build of longhorn-manager with hard-coded values.

**Comment by james-munson** [comment]
I think we're going to abandon this theory, as the list is too dynamic and likely always to be incomplete at best and misleading at worst.  Closing as "won't do."
